docs/src/app/components/PropTypeDescription.jsx

```diff
@@ -112,11 +112,28 @@ const PropTypeDescription = React.createClass({
         defaultValue = prop.defaultValue.value.replace(/\n/g, '');
       }
 
+      if (prop.required) {
+        key = `<span class="requiredProp">${key} \*</span>`;
+        requiredProps += 1;
+      }
+
+      if (prop.type.name === 'custom') {
+        if (getDeprecatedInfo(prop.type)) {
+          key = `~~${key}~~`;
+        }
+      }
+
       const description = generateDescription(prop.required, prop.description, prop.type);
 
       text += `| ${key} | ${generatePropType(prop.type)} | ${defaultValue} | ${description} |\n`;
     }
 
+    let requiredPropFootnote = (requiredProps === 1) ? '\\* required property' :
+      (requiredProps > 1) ? '\\* required properties' :
+        '';
+
+    text += `<div class="requiredPropFootnote">${requiredPropFootnote}</div>`;
```

Comment:
Given that the CSS is just a single attribute, I could just use an inline style anyway, but I can convert to jsx too. Whatever you prefer.